content/develop/tutorials/multipage-apps/dynamic-navigation.md

```diff
@@ -0,0 +1,422 @@
+---
+title: Create a dynamic navigation menu
+slug: /develop/tutorials/multipage/dynamic-navigation
+description: Streamlit makes it easy to build a custom navigation menu in your multipage app.
+---
+
+# Create a dynamic navigation menu
+
+`st.navigation` makes it easy to build dynamic navigation menus. You can change the set of pages passed to `st.navigation` with each rerun, which changes the navigation menu to match. This is a convenient feature for creating custom, role-based navigation menus.
+
+## Applied concepts
+
+- Use `st.navigation` and `st.Page` to define a multipage app.
+- Create a dynamic, role-based navigation menu.
+
+## Prerequisites
+
+- The following must be installed in your Python environment:
+
+  ```
+  streamlit>=1.36.0
+  ```
+
+- You should have a clean working directory called `your-repository`.
+- You should have a basic understanding of `st.navigation` and `st.Page`.
+
+## Summary
+
+In this example, we'll build a dynamic navigation menu for a multipage app that depends on the current user's role. You'll abstract away the use of username and creditials to simplify the example. Instead, you'll use a selectbox to let users choose a role and log in.
+
+The entrypoing file, `streamlit_app.py` will handle user authentication. The other pages will be stubs representing account management (`account/settings.py) and specific pages associated to three roles: Requester, Responder, and Admin. Requesters can access the account and request pages. Responders can access the account and respond pages. Admins can access all pages.
+
+Here's a look at what we'll build:
+
+<Collapse title="Complete code" expanded={false}>
+
+**Directory structure:**
+
+```
+your-repository/
+├── admin
+│   ├── admin_1.py
+│   └── admin_2.py
+├── images
+│   ├── horizontal_blue.png
+│   └── icon_blue.png
+├── request
+│   ├── request_1.py
+│   └── request_2.py
+├── respond
+│   ├── respond_1.py
+│   └── respond_2.py
+├── settings.py
+└── streamlit_app.py
+```
+
+**`streamlit_app.py`:**
+
+```python
+import streamlit as st
+
+if "role" not in st.session_state:
+    st.session_state.role = None
+
+DEFAULT_PAGES = {
+    None: None,
+    "Requester": "request/request_1.py",
+    "Responder": "respond/respond_1.py",
+    "Admin": "admin/admin_1.py",
+}
```

Comment:
Would it make sense to simplify this a bit? I'm not sure the dict + page comparison is adding a lot of value, and at least for me it took a bit longer to understand.

We could do:

```python
ROLES = ["Requester", "Responder", "Admin"]

# ...

respond_1 = st.Page(
    "respond/respond_1.py",
    title="Respond 1",
    icon=":material/healing:",
    default=(role == "Requester"),
)
```

^ At least for me, this is clearer and simpler to understand.

If you really prefer the current way I'm OK with it, but just wanted to flag that.

Reply:
I'm fine with that.